### ISSUE TITLE
Sonicwall Firewall Filters

### ISSUE THREAD

**Issue by satellite-no** [splunk/splunk-connect-for-syslog#1367]
Hey,
Would it be possible to add logs from Sonicwall firewalls? 

Closest TA is see is: https://splunkbase.splunk.com/app/4544

[sonicwall_anon.pcap.txt](https://github.com/splunk/splunk-connect-for-syslog/files/7735797/sonicwall_anon.pcap.txt)

**Comment by rfaircloth-splunk** [comment]
the pcap attached was corrupt can you attach a zip?

**Comment by satellite-no** [comment]
Shoot! I'll get another far you.  I actually have the app_parser built as well so I'll through that up with it.

**Comment by satellite-no** [comment]
Below is some test events from the sonicwall device along with a filter that is being used.

Events:
```
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=1024 m=97 n=1651190 src=192.69.254.11:60804:X3: dst=34.107.159.62:443:X1:62.159.107.34.bc.googleusercontent.com proto=tcp/https op=GET sent=4955 rcvd=6238 result=0 dstname=internal.home.luminatesec.com arg=/  code=64 Category="Not Rated" 
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683577 src=192.69.254.17:60571:X3:SIMILLANSWEEPER dst=10.69.21.172:135:X1: proto=tcp/135 NWG Arlington Heights  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683576 src=192.70.70.97:60714:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns NCEH Chicago  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683575 src=192.72.2.132:52890:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns IGG SSG  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683575 src=192.69.254.7:51635:X3:SIMILDC01 dst=10.248.10.30:53:X1: proto=udp/dns IGGTGGAL3L  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683575 src=192.70.70.97:56664:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns NCEH Chicago  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683575 src=192.72.2.150:63434:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns IGG SSG  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=1024 m=537 msg="Connection Closed" n=316039229 src=192.69.15.39:50026:X1: dst=10.69.254.7:445:X3:SIMILDC01 proto=tcp/445 sent=17542 rcvd=18318 vpnpolicy="ELGIN Lake Hills"  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683571 src=192.69.254.11:60804:X3: dst=34.107.159.62:443:X1:62.159.107.34.bc.googleusercontent.com proto=tcp/https   
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683570 src=192.72.2.132:54938:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns IGG SSG  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683569 src=192.69.254.7:53943:X3:SIMILDC01 dst=10.248.10.30:53:X1: proto=udp/dns IGGTGGAL3L  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683569 src=192.69.21.117:49293:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns NWG Arlington Heights  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683569 src=192.69.21.117:56906:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns NWG Arlington Heights  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683566 src=192.69.21.117:58038:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns NWG Arlington Heights  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683564 src=192.69.20.142:53447:X1: dst=192.168.9.72:9090:X1: proto=tcp/9090 PATHLAB Libertyville  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=1024 m=537 msg="Connection Closed" f=2 n=316039228 src=192.69.5.127:64181:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns sent=62 rcvd=137 vpnpolicy="GHA ST Charles"  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=1024 m=537 msg="Connection Closed" f=2 n=316039228 src=192.69.254.7:53036:X3:SIMILDC01 dst=10.248.10.30:53:X1: proto=udp/dns sent=83 rcvd=179 vpnpolicy="IGGTGGAL3L"  
<132>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=4 c=16 m=935 msg="IKE Responder: Peer's destination network does not match VPN policy's <b>Local Network</b>" n=1870292 src=50.247.147.145:500:: dst=202.238.53.81:500:: note="VPN Policy: FSS FEAE; Proposed network: 10.5.1.0/255.255.255.0" 
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683565 src=192.72.2.132:50861:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns IGG SSG  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=1024 m=537 msg="Connection Closed" n=316039228 src=192.70.70.153:52737:X1: dst=192.168.9.72:9090:X1: proto=tcp/9090 sent=208 vpnpolicy="NCEH Chicago"  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=262144 m=98 msg="Connection Opened" n=197683564 src=192.69.254.7:53578:X3:SIMILDC01 dst=10.248.10.30:53:X1: proto=udp/dns IGGTGGAL3L  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=1024 m=537 msg="Connection Closed" f=2 n=316039228 src=192.70.70.153:64715:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns sent=84 rcvd=157 vpnpolicy="NCEH Chicago"  
<132>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=4 c=16 m=88 msg="IKE Responder: IPSec proposal does not match (Phase 2)" n=3117352 src=50.247.147.145:500:: dst=202.238.53.81:500:: note="VPN Policy: FSS FEAE" 
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=16 m=352 msg="IKE Responder: Received Quick Mode Request (Phase 2)" n=3183233 src=50.247.147.145:500:: dst=202.238.53.81:500:: note="VPN Policy: FSS FEAE" 
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=1024 m=537 msg="Connection Closed" f=2 n=316039228 src=192.69.22.159:63808:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns sent=74 rcvd=154 vpnpolicy="ELG Elgin Main"  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=1024 m=537 msg="Connection Closed" n=316039228 src=192.69.21.153:54228:X1: dst=192.168.9.71:9090:X1: proto=tcp/9090 sent=208 vpnpolicy="NWG Arlington Heights"  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=1024 m=537 msg="Connection Closed" f=2 n=316039228 src=192.70.70.97:62801:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns sent=70 rcvd=126 vpnpolicy="NCEH Chicago"  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=1024 m=537 msg="Connection Closed" n=316039228 src=192.70.70.153:52738:X1: dst=192.168.9.71:9090:X1: proto=tcp/9090 sent=208 vpnpolicy="NCEH Chicago"  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=1024 m=537 msg="Connection Closed" f=2 n=316039228 src=192.69.22.172:37102:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns sent=60 rcvd=76 vpnpolicy="ELG Elgin Main"  
<134>id=firewall sn=C0EFE33057B0 time="2022-02-07 16:27:52 UTC" fw=202.238.53.81 pri=6 c=1024 m=537 msg="Connection Closed" f=2 n=316039228 src=192.69.22.159:61254:X1: dst=10.69.254.7:53:X3:SIMILDC01 proto=udp/dns sent=59 rcvd=134 vpnpolicy="ELG Elgin Main" 
```

Filter:
```
block parser dell_sonicwall-parser() {    
 channel {
        rewrite {
            r_set_splunk_dest_default(
                index("firewall")
                sourcetype("dell:sonicwall")
                vendor_product("dell_sonicwall")
            );    
            set("dell_sonicwall", value("fields.sc4s_syslog_format"));
            set("$(template t_hdr_msg)" value("MESSAGE"));
            unset(value("PROGRAM"));  
        };    


   };
};
application dell_sonicwall[sc4s-raw-syslog] {
	filter {                
    message('^\<\d+\>id=\S+\ssn\=\S+\stime=');
    };	

    parser { dell_sonicwall-parser(); };   
};
```
